subprojects/language-jvm/src/main/java/org/gradle/api/internal/tasks/compile/DefaultJvmLanguageCompileSpec.java

```diff
@@ -63,12 +65,12 @@ public void setTempDir(File tempDir) {
 
     @Override
     public FileCollection getSource() {
-        return source;
+        return ImmutableFileCollection.of(source);
     }
```

Comment:
This method is declared in an internal type, so we should be fine to change its return type to `Set<File>`.